packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -0,0 +1,321 @@
+
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+
+    @inject(KeybindingRegistry)
+    protected readonly keybindingRegistry: KeybindingRegistry;
+
+    @inject(KeymapsParser)
+    protected readonly keymapsParser: KeymapsParser;
+
+    @inject(KeymapsService)
+    protected readonly keymapsService: KeymapsService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    protected items: KeybindingItem[];
+
+    static readonly ID = 'keybindings.view.widget';
+
+    protected query: string = '';
+
+    protected readonly regexp = /<match>(.*?)<\/match>/g;
+    protected readonly keybindingSeperator = /<match>\+<\/match>/g;
+
+    protected readonly fuzzyOptions = {
+        pre: '<match>',
+        post: '</match>',
+    };
+
+    constructor() {
+        super();
+        this.id = KeybindingWidget.ID;
+        this.title.label = 'Keyboard Shortcuts';
+        this.title.closable = true;
+        this.update();
+    }
+
+    @postConstruct()
+    protected init(): void {
+        this.items = this.getItems();
+        if (this.keymapsService.onDidChangeKeymaps) {
+            this.keymapsService.onDidChangeKeymaps(() => {
+                this.searchKeybindings();
+                this.update();
+            });
+        }
+    }
+
+    protected searchKeybindings = () => {
+        this.items = [];
+        const searchField = this.findSearchField();
+        this.query = searchField ? searchField.value.trim().toLocaleLowerCase() : '';
+        const items = this.getItems();
+        items.forEach(item => {
+            const keys: (keyof KeybindingItem)[] = ['id', 'command', 'context', 'keybinding'];
+            let matched = false;
+            for (const key of keys) {
+                const string = item[key];
+                if (string) {
+                    const fuzzyMatch = fuzzy.match(this.query, string, this.fuzzyOptions);
+                    if (fuzzyMatch) {
+                        item[key] = fuzzyMatch.rendered;
+                        matched = true;
+                    }
+                }
+            }
+            if (matched) {
+                this.items.push(item);
+            }
+        });
+        this.update();
+    }
+
+    protected findSearchField(): HTMLInputElement | null {
+        return document.getElementById('search-kb') as HTMLInputElement;
+    }
+
+    protected render(): React.ReactNode {
+        return <div>
+            {this.renderSearch()}
+            {(this.items.length > 0) ? this.renderTable() : this.renderMessage()}
+        </div>;
+    }
+
+    protected renderSearch(): React.ReactNode {
+        return <div>
+            <div className='search-kb-container'>
+                <input id='search-kb'
+                    className={(this.items.length > 0) ? '' : 'no-kb'}
+                    type='text' placeholder='Search keybindings' onKeyUp={debounce(() => this.searchKeybindings(), 50)}></input >
```

Comment:
it creates a new debounce function each time and does not take any effect, you should store a function in the property, i.e
```ts
protected readonly searchKeybindings = debounce(() => this.doSearchKeybindings(), 50);
protected doSearchKeybindings(): void {
    ...
}
```